### PR TITLE
App

### PR DESCRIPTION
**Pull request by amDavid01**

### CODE REVIEW

File: src/api/api.ts [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+
+  if (data) {
+    options.body = JSON.stringify(data);
+  }
+
+  const url = `__client_url__${endpoint}`;
+
+  let response = await fetch(url, options);
+
+  if ([400, 401, 203, 500, 403].includes(response.status)) {
```
Comment (zpawn) [inline src/api/api.ts:495]: I see exchange refresh token only from `global_access_token.json`. But what if we safe token oBut during of using the app, the access and refresh tokens are exchanged and saved in the user state.  And then I see that you save only the access token. 

And you should also try to exchange the access token with the refresh token from the user store (not only from `global_access_token.json`)

File: src/components/DateField/DateField.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import "flatpickr/dist/themes/light.css";
+export type MappedFieldProps = DateTimePickerProps & {
+  id?: string;
+  label: string;
+  error?: boolean;
+  value: string;
+  onChange: (date: Date[]) => void;
+};
+
+export const DateField: FC<MappedFieldProps> = ({
```
Comment (zpawn) [inline src/components/DateField/DateField.tsx:16]: Use `DateInput` from `app-sdk`

File: src/components/Link/Contact.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      enabled: debouncedText.length > 2,
+    }
+  );
+
+  const contacts = contactsQuery.data?.value;
+
+  return (
+    <Stack gap={10} style={{ width: "100%" }} vertical>
+      <Stack vertical gap={6} style={{ width: "100%" }}>
+        <Input
```
Comment (zpawn) [inline src/components/Link/Contact.tsx:61]: use `Search` component from `app-sdk`

File: src/hooks/useQueryWithClient.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import {
+  QueryKey,
+  useMutation,
+  UseMutationOptions,
+  useQuery,
+  UseQueryOptions,
+  UseQueryResult,
+} from "@tanstack/react-query";
+
+export const useQueryWithClient = <
```
Comment (zpawn) [inline src/hooks/useQueryWithClient.ts:12]: use `useQueryWithClient` from `app-sdk`

File: src/hooks/useQueryMutationClient.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { IDeskproClient, useDeskproAppClient } from "@deskpro/app-sdk";
+import { useMutation, UseMutationOptions } from "@tanstack/react-query";
+
+export const useQueryMutationWithClient = <
```
Comment (zpawn) [inline src/hooks/useQueryMutationClient.ts:5]: use `useQueryMutationWithClient` from `app-sdk`

File: src/pages/Edit/Edit.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { useParams } from "react-router-dom";
+import { MutateObject } from "../../components/Mutate/Object";
+import { H1, Stack } from "@deskpro/deskpro-ui";
+
+export const EditObject = () => {
+  const { objectName, objectId } = useParams<{
+    objectName: "Contact";
+    objectId: string;
+  }>();
+  console.log(objectName, objectId);
```
Comment (zpawn) [inline src/pages/Edit/Edit.tsx:10]: remove `console`

File: src/styles.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import { H1 } from "@deskpro/deskpro-ui";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Link } from "react-router-dom";
+import styled, { css } from "styled-components";
+
+export const StyledLink = styled(Link)`
```
Comment (zpawn) [inline src/styles.ts:6]: use `Link` from `app-sdk`

File: src/components/Link/Contact.tsx (hunk 3 above)
Comment (amDavid01) [reply src/components/Link/Contact.tsx:61]: The input component looks alot better 
image
image

File: src/hooks/useQueryMutationClient.ts (hunk 5 above)
Comment (amDavid01) [reply src/hooks/useQueryMutationClient.ts:5]: not exported

File: src/styles.ts (hunk 7 above)
Comment (amDavid01) [reply src/styles.ts:6]: I prefer the styling on this. Plus there's nothing wrong with using your component

File: src/hooks/useQueryMutationClient.ts (hunk 5 above)
Comment (zpawn) [reply src/hooks/useQueryMutationClient.ts:5]: But you added this hook into `app-sdk`. Then please fix it

File: src/styles.ts (hunk 7 above)
Comment (zpawn) [reply src/styles.ts:6]: For what we created common ui components in the `app-sdk`?

File: src/components/Link/Contact.tsx (hunk 3 above)
Comment (zpawn) [reply src/components/Link/Contact.tsx:61]: ok

File: src/styles.ts (hunk 7 above)
Comment (amDavid01) [reply src/styles.ts:6]: Just because we created doesnt mean we need to use it. Same thing with the Search component

File: src/api/api.ts (hunk 1 above)
Comment (amDavid01) [reply src/api/api.ts:495]: no point in saving in the user state. they all use the exact same profile, set in the settings. We've discussed this flow before in dms and it is fine

File: src/components/DateField/DateField.tsx (hunk 2 above)
Comment (amDavid01) [reply src/components/DateField/DateField.tsx:16]: I am..
image

File: src/hooks/useQueryMutationClient.ts (hunk 5 above)
Comment (amDavid01) [reply src/hooks/useQueryMutationClient.ts:5]: Not sure why it was renamed to useMutation

File: src/styles.ts (hunk 7 above)
Comment (amDavid01) [reply src/styles.ts:6]: Refuses to redirect to the correct page.

File: src/components/DateField/DateField.tsx (hunk 2 above)
Comment (zpawn) [reply src/components/DateField/DateField.tsx:16]: I see 👌🏾

File: src/hooks/useQueryMutationClient.ts (hunk 5 above)
Comment (zpawn) [reply src/hooks/useQueryMutationClient.ts:5]: please, check it in the `app-sdk`, fix it if there is a bug and import it here

File: src/api/api.ts (hunk 1 above)
Comment (zpawn) [reply src/api/api.ts:495]: on line `510` - why do you save only the access token? refresh token won't be needed in the future?

Comment (amDavid01) [reply src/api/api.ts:495]: The default one works

Comment (zpawn) [reply src/api/api.ts:495]: How long does an access token live?

Comment (amDavid01) [reply src/api/api.ts:495]: 1 hour. the base refresh token works

**Pull request merged**